modules/admin-ui-frontend/app/scripts/shared/directives/passwordStrengthDirective.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+/**
+ * @ngdoc directive
+ * @description
+ * Checks the strength of a password.
+ *
+ * @example
+ * <pw-strength ng-model="model.password"></pw-strength>
+ */
+
+'use strict';
+
+angular.module('adminNg.directives')
+    .directive('pwStrength', [function () {
+      return {
+        restrict: 'E',
+        scope: {
+          password: '=ngModel'
+        },
+
+        link: function (scope, elem, attrs, ctrl) {
+          scope.$watch('password', function (pw) {
+
+            scope.rule =
+                requirements(pw && /[A-Z]/.test(pw)) +
+                requirements(pw && /[a-z]/.test(pw)) +
+                requirements(pw && /\d/.test(pw)) +
+                requirements(pw && /(?=.*\W)/.test(pw)) +
```

Comment:
I'm actually not entirely sure but isn't this just identical to

```suggestion
                requirements(pw && /\W/.test(pw)) +
```